floc.bs

```diff
@@ -1 +1,169 @@
+</section>
+
+<section>
+  <h2 id="cohort-assignment-algorithm">Cohort assignment algorithm</h2>
+  The browser could use machine learning algorithms to develop the [=interest cohort id=] to expose to a given document.
+
+  <h3 id="input-and-output">Input and output</h3>
+  The input features to the algorithm should be based on information from the browsing history, which may include the URLs, the page contents, or other factors.
+
+  The input features should be kept local on the browser and should not be uploaded elsewhere.
+
+  The output of the algorithm is the [=interest cohort id=].
+
+  <h3 id="privacy-guarantees">Privacy guarantees</h3>
+  The algorithm should have the following privacy properties. Sometimes generating an invalid [=interest cohort id=] may be helpful to meet these guarantees.
+
+  <h4 id="anonymity">Anonymity</h4>
+  The browser should ensure that the [=interest cohort ids=] are well distributed, so that each represents thousands of people, where a person is considered to be associated with an [=interest cohort id=] if that [=interest cohort id=] was recently computed for them. The browser may further leverage other anonymization methods, such as differential privacy.
+
+  <h4 id="no-browsing-history-recovering-from-cohorts">No browsing history recovering from cohorts</h4>
+  The browser should ensure that the [=interest cohort ids=] exposed to any given site does not reveal the browsing history.
+
+  <h4 id="no-sensitive-cohorts">No sensitive cohorts</h4>
+  The browser should ensure that the [=interest cohort ids=] are not correlated with <a href="#sensitive-information">sensitive information</a>.
+</section>
+
+<section>
+<h2 id=permissions-policy-integration>Permissions policy integration</h2>
+
+<p>This specification defines a [=policy-controlled feature=] identified by the string
+"<code><dfn id=interest-cohort-policy-controlled-feature>interest-cohort</dfn></code>". Its <a>default allowlist</a> is <code>*</code>.
+</section>
+
+<section>
+  <h2 id="privacy-considerations">Privacy considerations</h2>
+
+  <h3 id="permission">Permission</h3>
+  <h4 id="compute-eligibility">Eligibility for a page to be included in the interest cohort computation</h4>
+  By default, a page is eligible for the interest cohort computation if the {{Document/interestCohort()}} API is used in the page.
+
+  The page can opt itself out of the interest cohort computation through the "<code><a href="#interest-cohort-policy-controlled-feature">interest-cohort</a></code>" [=policy-controlled feature=]. [[!PERMISSIONS-POLICY]]
+
+  The user agent should offer a dedicated permission setting for the user to disallow sites from being included for interest cohort calculations.
+
+  <h4 id="access-eligibility">Permission to access the interest cohort</h4>
+  The page can restrict itself or subframes from accessing the interest cohort through the interest-cohort [=permissions policy=] header.
```

Comment:
The header doesn't work on subframes; for that you'd use `allow=""`. I'd suggest rephrasing this using the same phrasing as above, talking about the policy-controlled feature.

Reply:
Done.